app/services/ads/ad-slot-builder.js

```diff
@@ -4,6 +4,21 @@ import { getRenderComponentFor } from '../../utils/render-component';
 
 const MIN_ZEROTH_SECTION_LENGTH = 700;
 
+/**
+ * Check if element is too close `.affiliate-slot`
+ *
+ * @param {DOMElement} element
+ * @returns {boolean}
+ */
+function tooCloseToAffiliateSlots(element) {
+  const elementTop = offset(element).top || 0;
+
+  return Array.prototype.slice
+    .call(element.parentNode.querySelectorAll('.article-content > .affiliate-slot'))
+    .map(e => (offset(e).top || 0))
+    .some(top => (Math.abs(top - elementTop) < MIN_ZEROTH_SECTION_LENGTH));
```

Comment:
I would suggest using a different variable name here. `MIN_ZEROTH_SECTION_LENGTH` is used to control the distance of the first boxad from the top of the page. In this function the variable controls the distance from top of affiliate slot to h2 element. What do you think about `AFFILIATE_SLOT_HEIGHT` or sth similar?

Reply:
1f36d46